src/audio/google/google_rtc_audio_processing.c

```diff
@@ -95,6 +95,37 @@ struct google_rtc_audio_processing_comp_data {
 	struct comp_buffer *ref_comp_buffer;
 };
 
+#if CONFIG_GOOGLE_RTC_AUDIO_PROCESSING_MIC_BITS == 16
+typedef int16_t mic_sample_t;
+#define MIC_SCALE ((float)SHRT_MAX)
+#else
+typedef int32_t mic_sample_t;
+#define MIC_SCALE ((float)INT_MAX)
+#endif
+
+#if CONFIG_GOOGLE_RTC_AUDIO_PROCESSING_REF_BITS == 16
+typedef int16_t ref_sample_t;
+#define REF_SCALE ((float)SHRT_MAX)
+#else
+typedef int32_t ref_sample_t;
+#define REF_SCALE ((float)INT_MAX)
+#endif
+
+static inline float mic_to_float(mic_sample_t x)
```

Comment:
this doesn't look like the right path to enable all formats at runtime, we should just support both and toggle at runtime.

Reply:
Yeah, the formats themselves are build time.  Though there's no reason that can't be done with a function pointer.  Will fix.